src/utils/base64.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Punica - LwM2M server with REST API
+ * Copyright (C) 2018 8devices
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <nettle/base64.h>
+#include <string.h>
+
+#include <stdio.h>
+
+#include "base64.h"
+
+size_t base64_encoded_length(size_t data_length)
+{
+    return BASE64_ENCODE_RAW_LENGTH(data_length) + 1;
```

Comment:
Null terminator should not be included in the calculation, because it associates and limits the functions usage to null-terminated strings. We may want/need to use the base64 encoded/decoded strings in other scenarios, such as in the middle of the url: `http://localhost/foo?data=YmFy&count=3`. Library functions should only be concerned with the ACTUAL length calculation and encoding/decoding; it should be the responsibility of the library user to account for terminators.